marklogic-data-hub/src/main/resources/ml-modules/root/marklogic-data-hub-spark-connector/writeRecords.sjs

```diff
@@ -32,13 +32,16 @@ const headers = writeLib.buildHeaders(endpointConstants);
 
 const flowUtils = new FlowUtils();
 
+const uriTemplateRegEx = endpointConstants.uritemplate ? new RegExp("{([^}]+)}",'gi') : null;
```

Comment:
I don't think the endpoint should have knowledge of this - let generateUri check to see if uriTemplate is specified or not.

Reply:
I want to mention that if I do that RegExp inside the generateURI then we do that e.g. 200 times (or what the batch size is) instead of once

Reply:
Good point, I didn't catch that. I'd still like to keep the generateUri function args as "record, endpointConstant". What do you think of this:

- The endpoint will check for uriTemplate, and if it exists, it will add "uriTemplateRegex" to endpointConstants
- The generateUri function will check for "uriTemplateRegex", then for "uriTemplate" (and if uriTemplate exists, it will construct a regex for it) 

That allows for the optimization without complicating the interface of generateUri. Thoughts?

Reply:
This is an excellent idea. Ready for commit